core-image.html

```diff
@@ -61,24 +61,41 @@
 -->
 
 <link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../layout/layout.html">
+<dom-module id="core-image">
+  <style>
+    core-image {
+      display: inline-block;
+      overflow: hidden;
+      position: relative;
+    }
 
-<polymer-element name="core-image">
-<template>
-  <link rel="stylesheet" href="core-image.css">
-  <template if="{{!sizing}}">
-    <img id="img">
-  </template>
-  <template if="{{preload && fade}}">
-    <div id="placeholder" fit></div>
+    core-image[sizing] #img {
+      display: none;
+    }
+
+    core-image #placeholder {
+      background-color: inherit;
+      opacity: 1;
+    }
+
+    core-image #placeholder.faded-out {
+      transition: opacity 0.5s linear;
+      opacity: 0;
+    }
+  </style>
+  <template>
+    <img id="img" role="none" class$="{{imageClassName}}">
```

Comment:
Blank `src` on `img` is [bad](http://www.nczonline.net/blog/2009/11/30/empty-image-src-can-destroy-your-site/). Now that it's not wrapped in conditional rendering, this might be problematic.